### PR TITLE
Skip Audits (2)

### PR DESCRIPTION
**Pull request by Piinks**
## Description

I am auditing the codebase for skips that are associated with closed issues, lacking context, or are no longer relevant.

Second batch of skip audits, this brings some from https://github.com/flutter/flutter/pull/53616 so that PR could be more concise about a11y demos.

If any of these skips are still accurate, they will need an issue/comment attributed to them.

## Related Issues

#53707 
#19540
https://github.com/flutter/flutter/issues/54221

## Tests

Un-skipped, or properly referenced. :)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/foundation/diagnostics_test.dart [hunk 1]
```diff
@@ -1225,14 +1225,21 @@ void main() {
...
 
-    expect(present.toString(), equals('onClick: Closure: () => void'));
+    expect(
+      present.toString(),
+      equals(
+        'onClick:\n'
+        '  Closure: () => void from: function onClick() {\n'
+        '        }',
+      ),
+    );
```
Comment (Piinks) [inline packages/flutter/test/foundation/diagnostics_test.dart:1235]:
```suggestion
    expect(present.toString(), equals('onClick: Closure: () => void'));
```

File: packages/flutter/test/foundation/diagnostics_test.dart [hunk 2]
```diff
@@ -1232,7 +1232,7 @@ void main() {
     expect(missing.isFiltered(DiagnosticLevel.info), isFalse);
     validateObjectFlagPropertyJsonSerialization(present);
     validateObjectFlagPropertyJsonSerialization(missing);
-  }, skip: isBrowser);
+  });
```
Comment (Piinks) [inline packages/flutter/test/foundation/diagnostics_test.dart:1235]:
```suggestion
  }, skip: isBrowser); // Reference needed
```

File: packages/flutter/test/foundation/diagnostics_test.dart [hunk 3]
```diff
@@ -1232,7 +1232,7 @@ void main() {
     expect(missing.isFiltered(DiagnosticLevel.info), isFalse);
     validateObjectFlagPropertyJsonSerialization(present);
     validateObjectFlagPropertyJsonSerialization(missing);
-  }, skip: isBrowser);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/foundation/diagnostics_test.dart:1235]: @jonahwilliams is there an issue or comment I can add here?

File: packages/flutter/test/foundation/service_extensions_test.dart [hunk 4]
```diff
@@ -716,12 +716,12 @@ void main() {
     expect(trace, contains('dart:core,Object,Object.\n'));
     expect(trace, contains('package:test_api/test_api.dart,::,test\n'));
     expect(trace, contains('service_extensions_test.dart,::,main\n'));
-  }, skip: isBrowser);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/foundation/service_extensions_test.dart:719]: @ferhatb is there an issue or comment I can add here?

File: packages/flutter/test/foundation/stack_frame_test.dart [hunk 5]
```diff
@@ -68,7 +68,7 @@ void main() {
       expect(frames.contains(StackFrame.stackOverFlowElision), true);
     }
     expect(overflowed, true);
-  }, skip: isBrowser);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/foundation/stack_frame_test.dart:71]: @dnfield is there an issue or comment I can add here?

It looks like you left yourself a note in the change that added this (#49629) 🤣  : 

> For future me: The browser test harness doesn't seem to be able to handle this, but the VM does.

Comment (dnfield) [reply packages/flutter/test/foundation/stack_frame_test.dart:71]: hah, sorry about that.  I think linking this to the PR is probably ok.

Unfortunately, the browser harness can't handle testing something that intentionally causes a stackoverflow - the whole harness just crashes.  @yjbanov @ferhatb @jonahwilliams might know if anything can be done about that (or know who would know).

Comment (ferhatb) [reply packages/flutter/test/foundation/stack_frame_test.dart:71]: Yes, the browser completely freezes in that case and never responds back.

File: packages/flutter/test/foundation/service_extensions_test.dart (hunk 4 above)
Comment (ferhatb) [reply packages/flutter/test/foundation/service_extensions_test.dart:719]: // Compilation trace is Dart VM specific and not supported in browsers.

File: packages/flutter/test/foundation/stack_frame_test.dart (hunk 5 above)
Comment (dnfield) [reply packages/flutter/test/foundation/stack_frame_test.dart:71]: So maybe just

> // The VM test harness can handle a stack overflow, but the browser cannot - running this test in a browser will cause it to become unresponsive.

File: packages/flutter/test/foundation/diagnostics_test.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter/test/foundation/diagnostics_test.dart:1235]: I think the closure serialization does not work the same on the web. I will take a look at this one tomorrow

File: packages/flutter/test/foundation/diagnostics_test.dart [hunk 6]
```diff
@@ -1232,7 +1232,7 @@ void main() {
     expect(missing.isFiltered(DiagnosticLevel.info), isFalse);
     validateObjectFlagPropertyJsonSerialization(present);
     validateObjectFlagPropertyJsonSerialization(missing);
-  }, skip: isBrowser);
+  }); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/foundation/diagnostics_test.dart:1235]:
```suggestion
  }, skip: isBrowser); // https://github.com/flutter/flutter/issues/54221
```

**Review by ferhatb — APPROVED** [review]

**Pull request merged**